src/server/handlers/launchpad.js

```diff
@@ -411,34 +412,60 @@ export const internalFindSnap = async (repositoryUrl) => {
   throw new PreparedError(404, RESPONSE_SNAP_NOT_FOUND);
 };
 
-const internalFindSnapsByPrefix = async (urlPrefix) => {
+const internalFindSnaps = async (owner, token) => {
+  const orgs = await internalListOrganizations(owner, token);
+  const owners = [owner].concat(orgs.map((org) => org.login));
+  const urlPrefixes = owners.map(getRepoUrlPrefix);
   const username = conf.get('LP_API_USERNAME');
-  const cacheId = getUrlPrefixCacheId(urlPrefix);
+  const cacheIds = urlPrefixes.map(getUrlPrefixCacheId);
+  const memcached = getMemcached();
   const lpClient = getLaunchpad();
+  let snaps;
+  let remainingPrefixes;
 
   try {
-    const result = await getMemcached().get(cacheId);
-    if (result !== undefined) {
-      return result.map((entry) => {
+    snaps = await Promise.all(
+      cacheIds.map((cacheId) => memcached.get(cacheId))
+    );
+    remainingPrefixes = zip(urlPrefixes, snaps)
+      .filter(([, snap]) => snap === undefined)
+      .map(([urlPrefix, ]) => urlPrefix);
+    if (!remainingPrefixes.length) {
+      return [].concat(...snaps).map((entry) => {
```

Comment:
No, I'm using this as a one-level flatten operator:

    > var foo = [[1, 2], [3, 4]];
    undefined
    > [].concat(...foo)
    [ 1, 2, 3, 4 ]
    > [...foo]
    [ [ 1, 2 ], [ 3, 4 ] ]

However, you've caused me to realise that this code is confusingly written (each element of `snaps` here is in fact an array of snaps) and possibly wrong in one place, so I'll tidy that up.